### ISSUE TITLE
DockerHost.fromEnv() not updated for Docker for Mac

### ISSUE THREAD

**Issue by johnflavin** [spotify/docker-client#440]
## Description

The method `DefaultDockerClient.fromEnv()` was updated to use the unix socket on Mac in #436. However, an almost identical method exists in `DockerHost.fromEnv()` and this was not updated in #436. This causes test `DockerHostTest.testFromEnv()` to fail when using Docker for Mac, connected via the unix socket with no environment variables.

## Full backtrace

```text
testFromEnv(com.spotify.docker.client.DockerHostTest)  Time elapsed: 0.001 sec  <<< FAILURE!
java.lang.AssertionError:
Expected: "unix:///var/run/docker.sock"
     but: was "localhost:2375"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at com.spotify.docker.client.DockerHostTest.testFromEnv(DockerHostTest.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
```

**Comment by johnflavin** [comment]
I am not familiar with the `DockerHost` object. It doesn't seem to be used anywhere in `docker-client` other than its own tests. Is it used by external tools built on top of `docker-client`? If not, can it just be removed?

**Comment by davidxia** [comment]
We use it in github.com/spotify/helios. But, yes it's not clear from just
looking at the docker-client project.

On Thu, May 19, 2016 at 12:49 PM, Flavin <notifications@github.com> wrote:

> I am not familiar with the DockerHost object. It doesn't seem to be used
> anywhere in docker-client other than its own tests. Is it used by
> external tools built on top of docker-client? If not, can it just be
> removed?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-client/issues/440#issuecomment-220384763>
>



-- 
David Xia
Software Engineer
dxia@spotify.com

**Comment by mattnworb** [comment]
addressed in #441
